Field config update code snippet for common settings

tjheffner opened the pull request.

tjheffner committed:
Create update-all-fields.php

tjheffner committed:
Create readme.md

tjheffner committed:
Update update-all-fields.php

tjheffner committed:
Update update-all-fields.php

mike-potter commented on field_operations/update-all-fields.php:
```diff
@@ -0,0 +1,31 @@
+
+  // Pop off the entity key & generic fields (nid, uuid, revision_log, etc).
+  $fields = array_slice($fields, 18, null, true);
+
```
Is "18" a magic number?  How would you do this more generically?

tjheffner replied:
18 is a magic number, because when I var_dump($fields), the first 18 results for node fields are:

```
[0]=>
  string(3) "nid"
  [1]=>
  string(4) "uuid"
  [2]=>
  string(3) "vid"
  [3]=>
  string(8) "langcode"
  [4]=>
  string(4) "type"
  [5]=>
  string(5) "title"
  [6]=>
  string(3) "uid"
  [7]=>
  string(6) "status"
  [8]=>
  string(7) "created"
  [9]=>
  string(7) "changed"
  [10]=>
  string(7) "promote"
  [11]=>
  string(6) "sticky"
  [12]=>
  string(18) "revision_timestamp"
  [13]=>
  string(12) "revision_uid"
  [14]=>
  string(12) "revision_log"
  [15]=>
  string(29) "revision_translation_affected"
  [16]=>
  string(16) "default_langcode"
  [17]=>
  string(4) "path"
```

I think there is a isBaseField() method somewhere that would probably be better to check against, but I was having trouble calling it in this case.

mike-potter commented on field_operations/update-all-fields.php:
```diff
@@ -0,0 +1,31 @@
+    $field_name = $keys[$key];
+    $bundles = $fields[$value]['bundles'];
+
+    foreach ($bundles as $bundle) {
```
Probably need a comment here to say something like "In this example, we will set each field instance to be optional.  Replace this with whatever specific code you need to change the field instances."

tjheffner committed:
Update update-all-fields.php

tjheffner commented:
Updated to use the generic $entityFieldManager->getBaseFieldDefinitions('node'); instead of an arbitrary array_slice. Added a comment to the bundle loop that changes the setting itself.
